test/unittests/evm_state_test.cpp

```diff
@@ -177,6 +177,27 @@ TEST_F(evm_state, balance)
     EXPECT_EQ(result.output_data[5], 0x01);
 }
 
+TEST_F(evm_state, selfbalance)
+{
+    accounts[msg.destination].set_balance(0x0504030201);
+    auto code = bytecode{} + push(1) + OP_SELFBALANCE + mstore(0) + ret(32 - 6, 6);
```

Comment:
There's no argument to selfbalanace.

Reply:
But when `OP_SELFBALANCE` is undefined, there is an argument missing to `mstore(0)`.

Reply:
But shouldn't it just abort because of invalid instruction?

Reply:
Only in naive implementation of EVM. Here we check stack requirements for the whole block of instructions before executing them.

Reply:
I get that, but the concept is wrong then. evmone fails with a strange error code on invalid input then?

Reply:
Not sure this concept you mean, but yes, it can return other error code.

Reply:
This suggest to me that evmone cannot handle badly formatted EVM input, which has a stack underflow.